src/main/java/seedu/address/logic/commands/DeleteStudentCommand.java

```diff
@@ -44,14 +71,56 @@ public DeleteStudentCommand(Name name) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+        ObservableList<Student> studentList = model.getFilteredStudentList();
+        ArrayList<Student> listToCheck = new ArrayList<>();
+
+        for (Student student : studentList) {
+            if (student.getName().equals(name)) {
+                listToCheck.add(student);
+            }
+        }
```

Comment:
I think wee kean wrote a new method for the ModelManager which does this functionality. So perhaps you might want to make the change to that eventually to reduce spaghetti code. 👍